### PR TITLE
CAMEL-18131 - Adding health check to test conectivity

### PR DESCRIPTION
**Pull request by rhuan080**
Signed-off-by: Rhuan Rocha <rhuan080@gmail.com>

- [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
- [x] Each commit in the pull request should have a meaningful subject line and body.
- [x] If you're unsure, you can format the pull request title like `[CAMEL-XXX] Fixes bug in camel-file component`, where you replace `CAMEL-XXX` with the appropriate JIRA issue.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Run `mvn clean install -Psourcecheck` in your module with source check enabled to make sure basic checks pass and there are no checkstyle violations. A more thorough check will be performed on your pull request automatically.
Below are the contribution guidelines:
https://github.com/apache/camel/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: components/camel-aws/camel-aws2-s3/src/main/java/org/apache/camel/component/aws2/s3/AWS2S3ComponentVerifierExtension.java [hunk 1]
```diff
@@ -20,14 +20,7 @@
...
-import org.apache.camel.component.extension.verifier.ResultErrorBuilder;
 import org.apache.camel.component.extension.verifier.ResultErrorHelper;
-import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
-import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
-import software.amazon.awssdk.core.exception.SdkClientException;
-import software.amazon.awssdk.regions.Region;
-import software.amazon.awssdk.services.s3.S3Client;
-import software.amazon.awssdk.services.s3.S3ClientBuilder;
 
 public class AWS2S3ComponentVerifierExtension extends DefaultComponentVerifierExtension {
```
Comment (oscerd) [inline components/camel-aws/camel-aws2-s3/src/main/java/org/apache/camel/component/aws2/s3/AWS2S3ComponentVerifierExtension.java:25]: I think we could also remove the class

**Review by oscerd — APPROVED** [review]

**Comment by davsclaus** [comment]
Yes we can remove the component verifier which also means it should be removed on the component at
```
registerExtension(new AWS2S3ComponentVerifierExtension());
```

**Review by davsclaus — APPROVED** [review]

**Review by orpiske — APPROVED** [review]

**Comment by davsclaus** [comment]
The PR keeps failing, wonder what that error is?

Also mind that this essentially makes the S3 component do health-checks with a live check to AWS every time its invoked. Not sure if AWS would rate limit you - meaning that the interval for triggering health checks should be a reasonable (5 or 10s or longer)

**Comment by rhuan080** [comment]
@davsclaus I have tested it in my local environment and it works fine. I think it is some issue about the environment. Mainly because the issue is in another component that is not related to this update.

**Comment by rhuan080** [comment]
Hi @orpiske,

Based on the file changed list, the files updated were just about the aws2-s3. Thus, I think I don`t need to rebase it. However, I will do that.

**Comment by orpiske** [comment]
> Hi @orpiske,
> 
> Based on the file changed list, the files updated were just about the aws2-s3. Thus, I think I don`t need to rebase it. However, I will do that.

Screenshot 2022-08-30 at 19 16 51

Just as a matter of explanation: the `rebase` from upstream syncs your branch with the other changes that had been merged to `upstream/main`. Looking at your branch right, it was 227 commits behind `upstream/main`. When our diff script runs to find out what has changed, it compares the PR changes to `upstream/main`, therefore the greater the distance from main, the more will be tested (and this is why the label bot adds tags for jms, kafka and others even if the PR hasn't changed those).  

On a side note: I should probably add this to the contribution guide 🫣

**Pull request closed without merging**